src/test/java/io/vertx/core/http/HttpTest.java

```diff
@@ -5913,4 +5914,88 @@ private void testClientRequestWithLargeBodyInSmallChunks(boolean chunked) throws
     }));
     await();
   }
+
+
+  @Test
+  public void testProxyProtocolIllegalHeader() throws Exception {
```

Comment:
here we should use `waitFor(2)` and have two condition to end the test

Reply:
@vietj 

I have fixed all the test cases except this one.The problem with this one is that it behave differently for http1 and http2

For http1 

We get two HAProxyDecoder Exception
```
io.netty.handler.codec.haproxy.HAProxyProtocolException: io.netty.handler.codec.haproxy.HAProxyProtocolException: unknown identifier: GET
io.netty.handler.codec.haproxy.HAProxyProtocolException: io.netty.handler.codec.haproxy.HAProxyProtocolException: unknown identifier: host:
```

And for http2 

We get one HAProxyDecoder Exception

```
io.netty.handler.codec.haproxy.HAProxyProtocolException: header length (over 269) exceeds the allowed maximum (108)
```

And one from SSLHandler
```
java.nio.channels.ClosedChannelException
```

So if using waitFor(2) assuming 1 is from complete of client  the other should countdown when exception instanceof HAProxyProtocolException

The we will have 3 countdowns for http1 and two countdowns for http2

How do you suggest we should handle this problem?

Split the test between http1 / http2 ?

Reply:
that's fine for now